adaptivemd/condition.py

```diff
@@ -76,6 +76,9 @@ def __invert__(self):
     def __bool__(self):
         return self()
 
+    def __nonzero__(self):
```

Comment:
Could someone briefly explain what this function does and why it is needed here?

Reply:
This is for compatibility between python 2 and python 3. In 2, bool(x) will look for a method x.__nonzero__ , and x.__bool__ in 3. I have both there to evaluate properly in all cases. I thought it was nice to have __nonzero__ call __bool__ , so upon the eventual deprecation of python 2 this will just work as-is.

Reply:
Please document all functions in the code. Do you know which doc style we are using? Have a look at PyEMMA.